packages/auto-label/src/auto-label.ts

```diff
@@ -249,15 +245,102 @@ handler.autoLabelOnPR = async function autoLabelOnPR(
       logger.info(
         `Language label added to PR #${pull_number} in ${owner}/${repo} is ${language_label}`
       );
+      new_labels.push(language_label);
+    }
+  }
+
+  // Update all labels gathered so far if any
+  if (new_labels.length) {
+    logger.info(
+      `Labels added to PR# ${pull_number} in ${owner}/${repo} are ${JSON.stringify(
+        new_labels
+      )}`
+    );
+    await context.octokit.issues.addLabels({
+      owner,
+      repo,
+      issue_number: pull_number,
+      labels: new_labels,
+    });
+  }
+};
+
+/**
+ * Function updating a stale label on pull request based on configuration values.
+ * Removes previous staleness label if exists
+ */
+async function updateStalenessLabel(
+  context: Context<'pull_request'> | Context<'issues'>,
+  owner: string,
+  repo: string,
+  config: Config
+) {
+  // If user has turned on stale labels by configuring {staleness: {pullrequest: true, old: 60, critical: 120}}
+  // By default, this feature is turned off
+  if (!config.staleness?.pullrequest) {
+    return;
+  }
+  const response = await context.octokit.rest.pulls.list({
```

Comment:
By derfault, this will only return the first 25 pull requests, a better approach could be to use an async itetertor, like this:

```js
for await (const response of octokit.paginate.iterator(
  octokit.rest.issues.listForRepo,
  {
    owner: "octokit",
    repo: "rest.js",
  }
)) {
  // do whatever you want with each response, break out of the loop, etc.
}
```

Then you will automatically page through all pull requests.

Reply:
Yeah, I was thinking to isolate a call to pull requests only since correct me if I wrong, but call to `octokit.rest.issues.listForRepo` will return all issues in repo, including pull requests. We have already call to `context.octokit.issues.listForRepo.endpoint.merge` inside the `'schedule.repository'` event - I can patch it to process 5 issues counted by `labelWasNotAddedCount` and then only process pull requests when `labelWasNotAddedCount > 5` - what do you think?

Reply:
I don't think we need to optimize yet, we have `12,000` events an hour to work with, and we're only going to be enabling this on a few repos to start.

I would keep processing PRs in your own helper, and just switch to an async iterator, so you process them all.

Reply:
OK, thanks